Source/FakeItEasy.Specs/ConstructorWithVirtualCallsSpecs.cs

```diff
@@ -65,38 +92,165 @@ public class when_faking_a_class_which_calls_virtual_member_in_constructor_withi
+    }
 
-        It should_return_the_configured_value_from_the_scope_inside_and_outside_of_the_constructor_call = () =>
+    public class when_configuring_a_method_called_by_a_constructor_to_be_strict
+    {
+        static Exception exception;
+
+        Because of = () => exception = Record.Exception(() => A.Fake<MakesVirtualCallInConstructor>(options => options.Strict()));
+
+        private It should_throw_an_exception_during_the_constructor = () =>
         {
-            fake.VirtualMethodValueDuringConstructorCall.Should().Be("configured value in fake scope");
-            virtualMethodValueInsideOfScope.Should().Be("configured value in fake scope");
+            var expectedMessage = new[]
+            {
+                string.Empty,
+                "  Failed to create fake of type \"FakeItEasy.Specs.MakesVirtualCallInConstructor\".",
+                string.Empty,
+                "  Below is a list of reasons for failure per attempted constructor:",
+                "    No constructor arguments failed:",
+                "      No usable default constructor was found on the type FakeItEasy.Specs.MakesVirtualCallInConstructor.",
+                "      An exception of type FakeItEasy.ExpectationException was caught during this call. Its message was:",
+                "      Call to non configured method \"VirtualMethod\" of strict fake.",
+                "*"
+            }.AsTextBlock();
+
+            exception
+                .Should()
+                .BeAnExceptionOfType<FakeCreationException>()
+                .WithMessage(
+                    expectedMessage);
         };
+    }
 
-        It should_return_default_value_outside_of_scope = () => 
-            virtualMethodValueOutsideOfScope.Should().Be(string.Empty);
+    // This spec proves that we can configure methods of a strict fake and it applies to calls in the constructor (the 
+    // counterpart to the previous scenario with an un-configured strict fake):
+    public class when_configuring_a_method_called_by_a_constructor_to_be_strict_and_adding_configuration
+    {
+        static MakesVirtualCallInConstructor fake;
+
+        Because of = () =>
+            fake = A.Fake<MakesVirtualCallInConstructor>(options => options
+                .Strict()
+                .ConfigureFake(f => A.CallTo(() => f.VirtualMethod(A<string>._))
+                    .Returns("configured value of strict fake")));
+
+        It should_return_the_configured_value_during_the_constructor =
+            () => fake.VirtualMethodValueDuringConstructorCall.Should().Be("configured value of strict fake");
+
+        It should_return_the_configured_value_after_the_constructor =
+            () => fake.VirtualMethod(null).Should().Be("configured value of strict fake");
     }
 
-    public class when_faking_a_class_which_calls_virtual_property_members_in_constructor
+    public class when_wrapping_an_object_in_a_fake_that_calls_virtual_member_in_constructor

```

Comment:
No worries, "feature intersection tests" was clear to me.

As you point out, they do test the interaction between two features or aspects of FakeItEasy, which causes a problem. Where should tests of interacting features live? With tests for one feature? For the other?

They could move somewhere else, but I don't think "ConfigurationOptionsSpec" is necessarily the answer for all, although some would be a good fit. But I'd be more inclined to name the spec closer to the interface involved, maybe FakeOptionsBuilderSpecs. Perhaps that's too close to the implementation and not how the user would think of it, though. Still, these specs aren't user docs.

Truth be told, "ConstructorWithVirtualCallsSpecs" never really sat _that_ well with me as a concept, as in and of itself, a constructor with virtual calls isn't something that's part of FakeItEasy. Sure, it's important to know how we deal with those things, but I would've approached it from a "feature" direction, rather than a "property of the thing we're faking" direction. However, I didn't make a fuss because
1. I'm new to the whole Acceptance Test (or Spec) milieu, and 
2. grouping these classes is sort of like naming them, and that is one of the [Two Hard Things](http://martinfowler.com/bliki/TwoHardThings.html).

However, now that we've opened the discussion, I might as well propose a full-blown alternative, for comment. Were I to start moving things around on my own, I might do something like this (some classes would need to be renamed, and some _existing_ test classes might move, for example `when_asserting_must_have_happened_when_did_not_happen`):

- FakeConfiguratorSpecs
    - when_configuring_a_method_called_by_a_constructor_that_is_also_configured_by_a_fake_configurator
- FakeOptionsBuilderSpecs
    - when_configuring_a_method_called_by_a_constructor_to_be_strict
    - when_configuring_a_method_called_by_a_constructor_to_be_strict_and_adding_configuration
    - when_configuring_a_method_called_by_a_constructor_to_call_base_method
    - when_configuring_a_method_called_by_constructor_and_then_wrapping_an_object
    - when_faking_a_class_which_calls_configured_virtual_member_in_constructor
    - when_faking_a_class_whose_first_constructor_fails
    - when_wrapping_an_object_and_then_configuring_a_method_called_by_constructor
    - when_wrapping_an_object_in_a_fake_that_calls_virtual_member_in_constructor
    - when_wrapping_an_object_with_a_fake_that_has_a_fake_configurator (a tough one, as there's definite intersection here, but I picked the option builder specs because that's what's _visible_ to the user)
- FakeScopeSpecs?
    - when_configuring_a_method_called_by_a_constructor_from_within_a_scope
- UnconfiguredFakeSpecs?  I thought this was also tough.The virtual calls in constructor do tie these methods together, but I think the feature we're testing here is how an otherwise unconfigured fake behaves. 
    - when_faking_a_class_which_calls_virtual_member_in_constructor
    - when_faking_a_class_which_calls_virtual_properties_in_constructor

I think this make sense, but that could just be because everyone's ideas seem best to _them_.